### DISCUSSION TITLE
Code: FUNCTION_INVOCATION_TIMEOUT

### DISCUSSION THREAD

**Discussion by SrinivasReddyBobba** [vercel/community#4471] [Help] [answered]
### Page to Investigate

https://pre-production.adsgoat.in/

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/147658687/5d93ebba-4f01-4a77-9b58-5e092764b3c7)

**Comment by alibad** [comment]
I spent hours on this, and I am realizing I'm getting a timeout for no reason. It loads in a few seconds locally, and it gives me a 504 in production without any of the console.logs I have in "getServerSideProps" showing!

https://catalog.handstandquest.com/workouts/ai-opNb5ImoxKGKv3rmjLfn

**Comment by amyegan** [comment, marked as answer]
[504: FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#504:-function_invocation_timeout) means the function reached the [maximum duration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) limit. If the project is on a Hobby account, the limit is 10 seconds. The default limit on a Pro account is 15 seconds, but it can be  increased it up to 5 minutes.

**Comment by alibad** [comment]
So I was including AceEditor in the page. It worked great locally, but for some reason, it caused a 10.01 timeout in production without any hint of logging or errors. I have to say, this is one of the worst dev experiences I've had. I ended up just removing it and using a MUI component so I can move on.

const AceEditor = dynamic(() => import('react-ace').then((ace) => {
    import('ace-builds/src-noconflict/mode-json');
    import('ace-builds/src-noconflict/theme-solarized_light');
    return ace;
}), { ssr: false });
